code/modules/mob/living/simple_animal/hostile/giant_spider.dm

```diff
@@ -433,30 +447,31 @@
 		animal_owner.stop_automated_movement = TRUE
 
 	if(do_after(owner, wrap_time, target = to_wrap, interaction_key = INTERACTION_SPIDER_KEY))
-		var/obj/structure/spider/cocoon/casing = new(to_wrap.loc)
-		if(isliving(to_wrap))
-			var/mob/living/living_wrapped = to_wrap
-			// if they're not dead, you can consume them anyway
-			if(ishuman(living_wrapped) && (living_wrapped.stat != DEAD || !HAS_TRAIT(living_wrapped, TRAIT_SPIDER_CONSUMED)))
-				var/datum/action/innate/spider/lay_eggs/enriched/egg_power = locate() in owner.actions
-				if(egg_power)
-					egg_power.charges++
-					egg_power.UpdateButtons()
-					owner.visible_message(
-						span_danger("[owner] sticks a proboscis into [living_wrapped] and sucks a viscous substance out."),
-						span_notice("You suck the nutriment out of [living_wrapped], feeding you enough to lay a cluster of enriched eggs."),
-					)
-
-				living_wrapped.death() //you just ate them, they're dead.
-			else
-				to_chat(owner, span_warning("[living_wrapped] cannot sate your hunger!"))
-
-		to_wrap.forceMove(casing)
-		if(to_wrap.density || ismob(to_wrap))
-			casing.icon_state = pick("cocoon_large1", "cocoon_large2", "cocoon_large3")
-
+		wrap_target(to_wrap)
 	if(istype(animal_owner))
-		animal_owner.stop_automated_movement = TRUE
+		animal_owner.stop_automated_movement = FALSE
+
+/datum/action/cooldown/wrap/proc/wrap_target(atom/movable/to_wrap)
+	var/obj/structure/spider/cocoon/casing = new(to_wrap.loc)
+	if(isliving(to_wrap))
+		var/mob/living/living_wrapped = to_wrap
+		// if they're not dead, you can consume them anyway
+		if(ishuman(living_wrapped) && (living_wrapped.stat != DEAD || !HAS_TRAIT(living_wrapped, TRAIT_SPIDER_CONSUMED)))
```

Comment:
Is TRAIT_SPIDER_CONSUMED ever used anywhere? I don't see an add/remove trait anywhere

Reply:
Looks like you are right.
Looking at the commit history the intention for this trait seems to be that it marks consumed mobs so that you can only gain egg charges from them a single time (used to be a static list of mobs eaten), so I'll restore that behaviour.

Reply:
Nobody would have noticed this bug because of the fact that you could lay infinite eggs after eating one corpse anyway.